datumaro/components/media.py

```diff
@@ -741,44 +963,139 @@ class RoIImage(Image):
 
     def __init__(
         self,
-        data: Union[np.ndarray, Callable[[str], np.ndarray], None] = None,
-        *,
-        path: Optional[str] = None,
-        ext: Optional[str] = None,
         roi: BboxIntCoords,
-    ) -> None:
+        *args,
+        **kwargs,
+    ):
+        assert self.__class__ != RoIImage, (
+            f"Directly initalizing {self.__class__.__name__} is not supported. "
+            f"Please use a fractory function '{self.__class__.__name__}.from_image()'. "
+        )
+
         assert len(roi) == 4 and all(isinstance(v, int) for v in roi)
         self._roi = roi
         _, _, w, h = self._roi
-        super().__init__(data=data, path=path, ext=ext, size=(h, w))
+        super().__init__(size=(h, w), *args, **kwargs)
+
+    def as_dict(self) -> Dict[str, Any]:
+        attrs = super().as_dict()
+        attrs.pop("size", None)
+        return attrs
 
     @classmethod
-    def create_from_image(cls, img: Image, roi: BboxIntCoords) -> RoIImage:
-        if not isinstance(img, Image):
-            raise TypeError(f"type(img)={type(img)} should be Image.")
+    def from_file(cls, *args, **kwargs):
+        raise NotImplementedError
```

Comment:
Can we provide why we block this as follow? Same as elsewhere

```suggestion
    def from_file(cls, *args, **kwargs):
        raise DatumaroError("Please use a fractory function '{self.__class__.__name__}.from_image()'.")
```

Reply:
Fixed in https://github.com/openvinotoolkit/datumaro/pull/921/commits/08ce5909963ce28eacb7bbbd7a25c255acaeae7b